JavaScript/JavaScriptSDK/Logging.ts

```diff
@@ -81,5 +83,48 @@
                 }
             }
         }
+        
+        /**
+         * Resetting the throttle limits for Internal events
+         */
+        public static resetInternalEventsThrottle(): void {
+            this._eventCount = 0;
+        }
+
+        /**
+         * Sets the limit for the number of internal events before they are throttled
+         */
+        public static setMaxAllowedInternalThrottleLimit(limit: number): void {
+            if (!limit) {
+                return;
+            }
+            
+            this.MAX_ALLOWED_EVENT_LIMIT = limit;
+        }
+        
+        /**
+         * Throttles the internal logs based on the SDK configurations
+         */
+        private static _throttle(severity: LoggingSeverity, message: string, throttleLimit: number): void {
+            // If the event count exceeds the throttle limit, do nothing.
+            if (this._eventCount >= this.MAX_ALLOWED_EVENT_LIMIT) {
+                return;
+            }
+
+            // Push the event in the internal queue
+            if (_InternalLogging.verboseLogging() || severity === LoggingSeverity.CRITICAL) {
+                if (this.queue.length < this.MAX_QUEUE_SIZE) {
+                    this.queue.push(message);
+                    this._eventCount++;
+                }
+            }
+
+            // When throttle limit reached, send a special event
+            if (this._eventCount == this.MAX_ALLOWED_EVENT_LIMIT) {
```

Comment:
To clarify what I meant: It appears that the `throttleLimitMessage` is added to the  queue  when number of messages just reached  `MAX_ALLOWED_EVENT_LIMIT` and no messages were dropped due to throttling yet.